make/common/JdkNativeCompilation.gmk

```diff
@@ -35,6 +35,24 @@ include NativeCompilation.gmk
 # Hook to include the corresponding custom file, if present.
 $(eval $(call IncludeCustomExtension, common/JdkNativeCompilation.gmk))
 
+ifeq ($(OPENJDK_TARGET_OS), ios)
+  FindSrcDirsForLib = \
+    $(call uniq, $(wildcard \
+        $(TOPDIR)/src/$(strip $1)/$(OPENJDK_TARGET_OS)/native/lib$(strip $2) \
+        $(TOPDIR)/src/$(strip $1)/$(OPENJDK_TARGET_OS_TYPE)/native/lib$(strip $2) \
+        $(TOPDIR)/src/$(strip $1)/macosx/native/lib$(strip $2) \
+        $(TOPDIR)/src/$(strip $1)/share/native/lib$(strip $2)))
+endif
+
+ifeq ($(OPENJDK_TARGET_OS), android)
+  FindSrcDirsForLib = \
+    $(call uniq, $(wildcard \
+        $(TOPDIR)/src/$(strip $1)/$(OPENJDK_TARGET_OS)/native/lib$(strip $2) \
+        $(TOPDIR)/src/$(strip $1)/$(OPENJDK_TARGET_OS_TYPE)/native/lib$(strip $2) \
+        $(TOPDIR)/src/$(strip $1)/linux/native/lib$(strip $2) \
+        $(TOPDIR)/src/$(strip $1)/share/native/lib$(strip $2)))
+endif
+
```

Comment:
This needs some work. First of all, FindSrcDirsForLib should only be added to using +=, since the variable can have a prior value from the custom (closed) file. Secondly, later down,  FindSrcDirsForLib will have the "standard" directories appended, so no need to duplicate that here. Thirdly, order is important. If two files with the same name is present in the code base, only one will be selected, and the order of the directories in  FindSrcDirsForLib will determine the priorites. This code basically says "first ios, then unix, then macosx" or "first android, then unix, then linux". If this order is correct, then a better way to accomplish this is to just append macosx or linux *after* the normal assignment to  FindSrcDirsForLib below.

Reply:
Thanks for that remark, that makes sense. I updated the PR.
The problem is that on ios/android, we have 3 possible paths instead of the regular 2 (OPENJDK_TARGET_OS and OPENJDK_TARGET_OS_TYPE)
But your suggestion makes it much better to add the third path.